monkey/infection_monkey/master/ip_scanner.py

```diff
@@ -83,10 +83,8 @@ def _scan_addresses(
     def _scan_tcp_ports(
         self, ip: str, ports: List[int], timeout: float, stop: Event
     ) -> Dict[int, PortScanData]:
-        port_scan_data = {}
-
-        for p in interruptable_iter(ports, stop):
-            port_scan_data[p] = self._puppet.scan_tcp_port(ip, p, timeout)
+        interruptable_ports = interruptable_iter(ports, stop)
```

Comment:
Just pass the ports to `scan_tcp_ports()`. We can't know how this iterable will be used, and allowing something invisible to control the flow of execution inside the puppet could lead to bugs that are difficult to diagnose.

We can just remove `_scan_tcp_ports()` since it just wraps `IPuppet.scan_tcp_ports()`